### PR TITLE
limiting opening files with certain size

### PR DESCRIPTION
**Pull request by labkode**

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Brilliant - we totally failed to review/merge this properly :imp:

**Comment by labkode** [comment]
Hi @DeepDiver1975 

Yes I totally forgot about it.

I thing this check is really important for security related topics. Private mail

File: ajax/loadfile.php [hunk 1]
```diff
@@ -36,19 +36,26 @@
...
-	if ($encoding == "") {
-		// set default encoding if it couldn't be detected
-		$encoding = 'ISO-8859-15';
+	$maxsize = \OCP\Config::getSystemValue("max_size_text_editor", 4194304); // default of 4MB
+	$size = \OC\Files\Filesystem::filesize($path);
+	if($size > $maxsize) {
+		OCP\JSON::error(array('data' => array( 'message' => "Max size for opening files is $maxsize.")));
+	} else {
+		$filecontents = \OC\Files\Filesystem::file_get_contents($path);
+		$encoding = mb_detect_encoding($filecontents."a", "UTF-8, WINDOWS-1252, ISO-8859-15, ISO-8859-1, ASCII", true);
```
Comment (LukasReschke) [inline ajax/loadfile.php]: I'd prefer if we would just read the first 500/1000/whatever characters and guess the encoding of that one. - No need to do this over the whole file from my PoV.

Comment (LukasReschke) [inline ajax/loadfile.php]: … or just handle this completely via JS :speak_no_evil:

Comment (LukasReschke) [inline ajax/loadfile.php]: Let me see what http://ace.c9.io/api/edit_session.html#EditSession.setNewLineMode can do then…

Comment (DeepDiver1975) [inline ajax/loadfile.php]: unrelated with respect to this change

**Comment by DeepDiver1975** [comment]
superseded by https://github.com/owncloud/files_texteditor/pull/120

**Pull request closed without merging**